circus-train-gcp/src/test/java/com/hotels/bdp/circustrain/gcp/context/GCPBeanPostProcessorTest.java

```diff
@@ -16,46 +16,78 @@
 package com.hotels.bdp.circustrain.gcp.context;
 
 import static org.mockito.Matchers.anyString;
+import static org.mockito.Mockito.doNothing;
+import static org.mockito.Mockito.doReturn;
 import static org.mockito.Mockito.times;
 import static org.mockito.Mockito.verify;
 import static org.mockito.Mockito.when;
 
 import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.Path;
+import org.junit.Before;
 import org.junit.Test;
 import org.junit.runner.RunWith;
-import org.mockito.Spy;
+import org.mockito.Mock;
 import org.mockito.runners.MockitoJUnitRunner;
 
 import com.hotels.bdp.circustrain.context.CommonBeans;
+import com.hotels.bdp.circustrain.gcp.BindGoogleHadoopFileSystem;
+import com.hotels.bdp.circustrain.gcp.DistributedFileSystemPathProvider;
+import com.hotels.bdp.circustrain.gcp.FileSystemFactory;
+import com.hotels.bdp.circustrain.gcp.GCPCredentialCopier;
+import com.hotels.bdp.circustrain.gcp.GCPCredentialPathProvider;
+import com.hotels.bdp.circustrain.gcp.RandomStringFactory;
 
 @RunWith(MockitoJUnitRunner.class)
 public class GCPBeanPostProcessorTest {
-  private @Spy GCPSecurity security = new GCPSecurity();
-  private @Spy Configuration configuration = new Configuration();
+  private @Mock RandomStringFactory randomStringFactory;
+
+  private @Mock GCPSecurity security;
+  private @Mock GCPCredentialPathProvider credentialPathProvider;
+  private @Mock DistributedFileSystemPathProvider distributedFileSystemPathProvider;
+  private @Mock Configuration configuration;
+  private @Mock BindGoogleHadoopFileSystem bindGoogleHadoopFileSystem;
+  private @Mock FileSystemFactory fileSystemFactory;
+  private @Mock GCPCredentialCopier credentialCopier;
+
+  private GCPBeanPostProcessor processor;
+
+  @Before
+  public void init() {
+    doNothing().when(configuration).set(anyString(), anyString());
+    processor = new GCPBeanPostProcessor(credentialPathProvider, distributedFileSystemPathProvider,
+        bindGoogleHadoopFileSystem, fileSystemFactory, credentialCopier);
+  }
 
   @Test
   public void postProcessAfterInitializationWithCorrectBeanName() throws Exception {
     String beanName = CommonBeans.BEAN_BASE_CONF;
-    GCPBeanPostProcessor processor = new GCPBeanPostProcessor(security);
     processor.postProcessAfterInitialization(configuration, beanName);
-    verify(security, times(1)).getCredentialProvider();
+    verify(credentialPathProvider).newPath();
   }
 
   @Test
   public void postProcessAfterInitializationWithIncorrectBeanName() throws Exception {
     String beanName = "notBaseConf";
-    GCPBeanPostProcessor processor = new GCPBeanPostProcessor(security);
     processor.postProcessAfterInitialization(configuration, beanName);
-    verify(security, times(0)).getCredentialProvider();
+    verify(credentialPathProvider, times(0)).newPath();
   }
 
   @Test
   public void postProcessAfterInitializationWithBlankCredentialProviderDoesntModifyConfiguration() throws Exception {
```

Comment:
remove this test, your code doesn't modify the configuration so you should not have to test it, it's also using `security` which is probably a remnant of a previous refactoring.